modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts

```diff
@@ -62,7 +63,9 @@ function getValidFields({
 		const type = propertyValue.type;
 
 		if (type === EFieldType.ARRAY) {
-			return;
+			if (propertyValue.items && propertyValue.items.$ref) {
```

Comment:
nit: we can simplify this to `if (type === EFieldType.ARRAY && propertyValue.items?.$ref)`

Reply:
is this for scalar?

Reply:
This will allow only scalar arrays.